File: since2020/CodeForces/1324B.cpp
/**
 *  @judge CodeForces
 *  @id 1324B
 *  @name Yet Another Palindrome Problem
 *  @contest Codeforces Round #627
 * 
 *  @tag Palindrome
 */
#include <bits/stdc++.h>
using namespace std;

int n;
vector<int> a;
map<int, int> M;

bool solve() {
	M.clear();
	for (int i = 0; i < a.size(); i++) {
		if (!M[a[i]])
			M[a[i]] = i + 1;
		else if (i - M[a[i]] > 0)
			return true;
	}
	return false;
}

int main() {
	int t;
	for (cin >> t; t; t--) {
		cin >> n;
		a = vector<int>(n, 0);
		for (auto && x : a)
			cin >> x;
		cout << (solve() ? "YES" : "NO") << endl;
		a.clear();
	}
}

File: since2020/CodeForces/1324B.py
'''
	@judge CodeForces
	@id 1324B
	@name Yet Another Palindrome Problem
	@contest Codeforces Round #627
	
	@tag Palindrome
'''
from sys import stdin
from itertools import groupby

def solve(arr):
	return any(
		map(lambda p: max(p) - min(p) > 1,
			map(lambda ps: [ x for x, _ in ps[1] ],
				groupby(sorted(enumerate(arr), key=lambda p: p[1]), key=lambda p: p[1])
	)))

input()
for _, line in zip(*[ iter(stdin) ] * 2):
	arr = line.split()
	print('YES' if solve(arr) else 'NO')